SDWebImage/SDWebImageDownloaderOperation.m

```diff
@@ -231,25 +236,11 @@ - (void)done {
 }
 
 - (void)reset {
-    __weak typeof(self) weakSelf = self;
-    dispatch_barrier_async(self.barrierQueue, ^{
-        [weakSelf.callbackBlocks removeAllObjects];
-    });
+    LOCK(self.callbacksLock);
+    [self.callbackBlocks removeAllObjects];
+    UNLOCK(self.callbacksLock);
     self.dataTask = nil;
     
-    NSOperationQueue *delegateQueue;
```

Comment:
This is previouslly to keep that `imageData` mutable is thread-safe. However, if the `reset` method is called. This means the operation is neither cancelled or finished and will be removed from `SDWebImageDownloader`'s operation queue. So we do not need any work for the operation itself to let ARC dealloc it.